resources/function_help/json/count_substring

```diff
@@ -0,0 +1,31 @@
+{
+  "name": "count_substring",
+  "type": "function",
+  "groups": ["String"],
+  "description": "Counts the number of non-overlapping occurrences of a substring within a string.",
```

Comment:
> non-overlapping occurrences

is that actually the case?
Qt docs say:
> `int QString::count(const QString &str, Qt::CaseSensitivity cs = Qt::CaseSensitive) const`
> Returns the number of (potentially overlapping) occurrences of the string str in this string.

Could you maybe add a couple of tests for this?

Reply:
> > non-overlapping occurrences
> 
> is that actually the case? Qt docs say:
> 
> > `int QString::count(const QString &str, Qt::CaseSensitivity cs = Qt::CaseSensitive) const`
> > Returns the number of (potentially overlapping) occurrences of the string str in this string.
> 
> Could you maybe add a couple of tests for this?

Yes. That's how it works :). Don't know the inside details but I tested manually and now you can also see directly from the tests :). Thanks for reviewing.